gobblin-service/src/main/java/org/apache/gobblin/service/modules/dataset/ExternalDatasetDescriptor.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.gobblin.service.modules.dataset;
+
+import com.typesafe.config.Config;
+import com.typesafe.config.ConfigValueFactory;
+import java.io.IOException;
+import java.util.ArrayList;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.ToString;
+import org.apache.gobblin.service.modules.flowgraph.DatasetDescriptorConfigKeys;
+import org.apache.gobblin.service.modules.flowgraph.DatasetDescriptorErrorUtils;
+import org.apache.gobblin.util.ConfigUtils;
+
+
+/**
+ * Describes a external dataset not on HDFS
+ * e.g, https://some-api:443/user/123/names where the path is the full URI
+ */
+@ToString (exclude = {"rawConfig"})
+@EqualsAndHashCode (exclude = {"rawConfig"}, callSuper = true)
+public class ExternalDatasetDescriptor extends BaseDatasetDescriptor implements DatasetDescriptor {
+
+  @Getter
+  private final String path;
+  @Getter
+  private final Config rawConfig;
+
+  public ExternalDatasetDescriptor(Config config) throws IOException {
+    super(config);
+    // refers to the full HTTP url
+    this.path = ConfigUtils.getString(config, DatasetDescriptorConfigKeys.PATH_KEY, "");
```

Comment:
what does an empty path mean?  shouldn't it be required (not defaulting to `""`)?